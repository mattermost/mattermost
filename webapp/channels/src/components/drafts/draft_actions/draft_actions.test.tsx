// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';
import {Provider} from 'react-redux';

import mockStore from 'tests/test_store';

import DraftActions from './draft_actions';

describe('components/drafts/draft_actions', () => {
    const baseProps = {
        displayName: '',
        draftId: '',
        itemId: '',
        onDelete: jest.fn(),
        onEdit: jest.fn(),
        onSend: jest.fn(),
        canSend: true,
        canEdit: true,
        onSchedule: jest.fn(),
        channelId: '',
    };

    it('should match snapshot', () => {
        const store = mockStore();

        const wrapper = shallow(
            <Provider store={store}>
                <DraftActions
                    {...baseProps}
                />
            </Provider>,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
