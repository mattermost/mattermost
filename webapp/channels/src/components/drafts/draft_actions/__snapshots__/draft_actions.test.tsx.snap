// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/drafts/draft_actions should match snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Memo(DraftActions)
    canEdit={true}
    canSend={true}
    channelId=""
    displayName=""
    draftId=""
    itemId=""
    onDelete={[MockFunction]}
    onEdit={[MockFunction]}
    onSchedule={[MockFunction]}
    onSend={[MockFunction]}
  />
</ContextProvider>
`;
